demos/object_detection_demo_ssd_async/main.cpp

```diff
@@ -229,100 +281,93 @@ int main(int argc, char *argv[]) {
-            // Here is the first asynchronous point:
-            // in the async mode we capture frame to populate the NEXT infer request
-            // in the regular mode we capture frame to the CURRENT infer request
-            if (!cap.read(next_frame)) {
-                if (next_frame.empty()) {
-                    isLastFrame = true;  // end of video file
-                } else {
-                    throw std::logic_error("Failed to get frame from cv::VideoCapture");
-                }
-            }
-            if (isAsyncMode) {
-                if (isModeChanged) {
-                    frameToBlob(curr_frame, async_infer_request_curr, imageInputName);
-                }
-                if (!isLastFrame) {
-                    frameToBlob(next_frame, async_infer_request_next, imageInputName);
-                }
-            } else if (!isModeChanged) {
-                frameToBlob(curr_frame, async_infer_request_curr, imageInputName);
-            }
+        Presenter presenter(FLAGS_u, static_cast<int>(cap.get(cv::CAP_PROP_FRAME_HEIGHT)) - graphSize.height - 10,
+                            graphSize);
+        // -----------------------------------------------------------------------------------------------------
 
-            auto t1 = std::chrono::high_resolution_clock::now();
-            double ocv_decode_time = std::chrono::duration_cast<ms>(t1 - t0).count();
+        // --------------------------- 7. Do inference ---------------------------------------------------------
+        slog::info << "Start inference " << slog::endl;
 
-            t0 = std::chrono::high_resolution_clock::now();
-            // Main sync point:
-            // in the truly Async mode we start the NEXT infer request, while waiting for the CURRENT to complete
-            // in the regular mode we start the CURRENT request and immediately wait for it's completion
-            if (isAsyncMode) {
-                if (isModeChanged) {
-                    async_infer_request_curr->StartAsync();
-                }
-                if (!isLastFrame) {
-                    async_infer_request_next->StartAsync();
-                }
-            } else if (!isModeChanged) {
-                async_infer_request_curr->StartAsync();
+        std::cout << "To close the application, press 'CTRL+C' here or switch to the output window and "
+                     "press ESC or 'q' key" << std::endl;
+        std::cout << "To switch between min_latency/user_specified modes, press TAB key in the output window" 
+                  << std::endl;
+
+        while ((cap.isOpened()
+                || !completedRequestResults.empty()
+                || ((isUserSpecifiedMode && emptyRequests.size() < FLAGS_nireq)
+                    || (!isUserSpecifiedMode && emptyRequests.size() < 1)))
+               && callbackException == nullptr) {
+            if (callbackException) std::rethrow_exception(callbackException);
```

Comment:
You should protect some of these vars because they are modified by another thread. I suggest moving the condition under the scope of the following `lock_guard`.